build.py

```
@@ -3,45 +3,177 @@
+
+    def _scan_file(self, path: Path) -> set[str]:
+        """Extracts top-level module names from a single Python file."""
+        imports = set()
+        try:
+            with open(path, encoding="utf-8") as f:
+                for line in f:
+                    match = self.import_re.match(line.strip())
+                    if match:
+                        imports.add(match.group(1))
+        except Exception:
+            pass
```

Ruff (0.15.12)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)